src/tiledb/cloud/utilities/_common.py

```diff
@@ -272,3 +274,73 @@ def reader(stream):
             stderr = stderr_future.result()
 
             return stdout, stderr
+
+
+def batch(func: Callable) -> Callable:
+    """
+    Decorator to run a function as a batch UDF on TileDB Cloud.
+
+    :param func: function to run
+    """
+
+    def filter_kwargs(
+        function: Callable, kwargs: Mapping[str, Any]
+    ) -> Mapping[str, Any]:
+        """
+        Filter kwargs to only include valid arguments for the function.
+
+        :param function: function to validate kwargs for
+        :param kwargs: kwargs to filter
+        :return: filtered kwargs
+        """
+        valid_args = inspect.signature(function).parameters
+        filtered_kwargs = {k: v for k, v in kwargs.items() if k in valid_args}
+        return filtered_kwargs
```

Comment:
Since this is only used with `func` as the `function` parameter, I would recommend either just using the captured `func` value (i.e., `def filter_kwargs(kwargs: ...) -> ...`) or pulling this out into the module level (`def _filter_kwargs(function, kwargs)` at the root.

Reply:
I pulled it out to the root, in case we need to use it in the future.